pkgs/applications/virtualization/vmware-workstation/default.nix

```diff
@@ -0,0 +1,419 @@
+
+with lib;
+
+let
+  vmware-unpack-env = buildFHSUserEnv rec {
+    name = "vmware-unpack-env";
+    targetPkgs = pkgs: [ zlib ];
+  };
+  gdbm3 = gdbm.overrideAttrs ( old: rec {
+    version = "1.8.3";
+
+    src = fetchurl {
+      url = "mirror://gnu/gdbm/${old.pname}-${version}.tar.gz";
+      sha256 = "sha256-zDQDOKLii0AFirnrU1SiHVP4ihWC6iG6C7GFw3ooHck=";
+    };
+
+    installPhase = ''
+      mkdir -p $out/lib
+      cp .libs/libgdbm*.so* $out/lib/
+    '';
+
+  });
+in stdenv.mkDerivation rec {
+  pname = "vmware-workstation";
+  version = "16.2.3";
+  build = "19376536";
+
+  buildInputs = [ libxslt libxml2 libuuid gdbm3 readline xz bzip2 cups glibc libaio vulkan-loader alsa-lib libpulseaudio libGL numactl libX11 libXi kmod ];
+  nativeBuildInputs = [ python3 vmware-unpack-env autoPatchelfHook makeWrapper ]
+  ++ lib.optional (enableInstaller) sqlite;
+
+  src = fetchurl {
+    url = "https://download3.vmware.com/software/WKST-1623-LX-New/VMware-Workstation-Full-${version}-${build}.x86_64.bundle";
+    sha256 = "sha256-+JE1KnRfawcaBannIyEr1TNZTF7YXRYYaFMVq0/erbM=";
+  };
+
+  unpackPhase = ''
+    ${vmware-unpack-env}/bin/vmware-unpack-env -c "sh ${src} --extract unpacked"
+  '';
+
+  installPhase = ''
+
+    mkdir -p \
+      $out/bin \
+      $out/etc/vmware \
+      $out/etc/init.d \
+      $out/lib/vmware \
+      $out/share/doc
+
+    #### Replicate vmware-installer's order but VMX first because of appLoader
+
+    ${lib.optionalString (enableInstaller) ''
+      ## VMware installer
+      echo "Installing VMware Installer"
+      unpacked="unpacked/vmware-installer"
+
+      local vmware_installer_version=$(cat "unpacked/vmware-installer/manifest.xml" | grep -oPm1 "(?<=<version>)[^<]+")
+      dest="$out/lib/vmware-installer/$vmware_installer_version"
+
+      mkdir -p $dest
+
+      cp -r $unpacked/vmis* $dest/
+      cp -r $unpacked/sopython $dest/
+      cp -r $unpacked/python $dest/
+      cp -r $unpacked/cdsHelper $dest/
+      cp -r $unpacked/vmware* $dest/
+      cp -r $unpacked/bin $dest/
+      cp -r $unpacked/lib $dest/
+
+      chmod +x $dest/vmis-launcher
+      chmod +x $dest/sopython/*
+      chmod +x $dest/python/init.sh
+      chmod +x $dest/vmware-*
+
+      ln -s $dest/vmware-installer $out/bin/vmware-installer
+
+      mkdir -p $out/etc/vmware-installer
+
+      cat <<EOF > $out/etc/vmware-installer/bootstrap
+      VMWARE_INSTALLER="$dest"
+      VERSION="$vmware_installer_version" # For backwards compability
+      VMISVERSION="$vmware_installer_version"
+      VMISBUILDNUM="${build}"
+      VMISPYVERSION="39"
+      EOF
+
+      mkdir -p $out/etc/vmware-installer/components
+
+      # create database of vmware guest tools (avoids vmware fetching them later)
+      local database_filename=$out/etc/vmware-installer/database
+      echo -n "" > $database_filename
+      sqlite3 "$database_filename" "CREATE TABLE settings(key VARCHAR PRIMARY KEY, value VARCHAR NOT NULL, component_name VARCHAR NOT NULL);"
```

Comment:
I don't think we have that in nativeBuildInputs

Reply:
pkgs.sqlite specified in nativeBuildInputs with ++ lib.optional